### PR TITLE
[]NPU] Update adapter and backend to use new level zero ext table

### PR DESCRIPTION
**Pull request by XinWangIntel**
### Details:
 - *use ze_graph_dditable_ext_t to replace local ze_graph_dditable_ext_last_t*
 - *use ze_graph_ext_version_t as template parameter instead of ze_graph_dditable_ext_1_x*
 - *update commit of level-zero-ext to use one ddi table*

Plugin with current driver and driver released later, will use ze_graph_dditable_ext_t and ZE_extension_graph, the version is ZE_GRAPH_EXT_VERSION_CURRENT inside current level-zero-ext commit, 1.8 now

Plugin with old driver, will use ze_graph_dditable_ext_t to reinterpret old ddi table, use largest version supported by driver, , backend work on ze_graph_ext_version_t, compiler will work based on ze_graph_ext_version_t. Fpr example, largest ext version supported by driver is 1.7. plugin will use 1.7.

### Tickets:
 - *155313*

### CODE REVIEW

**Comment by XinWangIntel** [comment]
build_jenkins

File: src/plugins/intel_npu/src/backend/include/zero_types.hpp [hunk 1]
```diff
@@ -13,10 +13,6 @@
 
 #include "intel_npu/config/runtime.hpp"
 
-/**
- * @brief Last version of Table of Graph Extension functions used within plugin
- */
-using ze_graph_dditable_ext_last_t = ze_graph_dditable_ext_1_8_t;
```
Comment (pereanub) [inline src/plugins/intel_npu/src/backend/include/zero_types.hpp:19]: Just a quick question here. Are these changes only for the graph dditable? I mean, the command queue structure also has the name 1_0.

Comment (XinWangIntel) [reply src/plugins/intel_npu/src/backend/include/zero_types.hpp:19]: Just for ddi table, in https://github.com/intel/level-zero-npu-extensions/commit/a6487cc2c5da9aa13db9e005a320a1b6a0ee5919, just graph ext is impacted, other table is on level zero side.

File: src/plugins/intel_npu/src/compiler/src/driver_compiler_adapter.cpp [hunk 2]
```diff
@@ -41,46 +41,46 @@ LevelZeroCompilerAdapter::LevelZeroCompilerAdapter(std::shared_ptr<IEngineBacken
 
     switch (graphExtVersion) {
     case ZE_GRAPH_EXT_VERSION_1_3:
```
Comment (jwludzik) [inline src/plugins/intel_npu/src/compiler/src/driver_compiler_adapter.cpp:43]: What about Graph Extension 1.9? The plan is that driver pushes the change with 1.9 and drop the previous one from Driver extension properties list. It means that 1.8 is not expose any longer by driver and 1.9 is available together with 1.2 up to 1.7 graph extensions

Comment (PatrikStepan) [reply src/plugins/intel_npu/src/compiler/src/driver_compiler_adapter.cpp:43]: That change is not available in the graph-ext version we use in this PR. We agreed to leave that for another PR, where we would also fix the total_mem_size property.

Comment (jwludzik) [reply src/plugins/intel_npu/src/compiler/src/driver_compiler_adapter.cpp:43]: Thanks for explanation

**Review by PatrikStepan — APPROVED** [review]

File: src/plugins/intel_npu/src/backend/src/zero_backend.cpp [hunk 3]
```diff
@@ -30,7 +30,7 @@ uint32_t ZeroEngineBackend::getGraphExtVersion() const {
 }
 
 bool ZeroEngineBackend::isBatchingSupported() const {
-    return _instance->isExtensionSupported(std::string(ZE_GRAPH_EXT_NAME_1_6), ZE_MAKE_VERSION(1, 6));
```
Comment (csoka) [inline src/plugins/intel_npu/src/backend/src/zero_backend.cpp:33]: This needs to remain on ZE_GRAPH_EXT_NAME_1_6 for compatibility with old drivers

Comment (XinWangIntel) [reply src/plugins/intel_npu/src/backend/src/zero_backend.cpp:33]: Fixed, since the macro is not in current level-zero-ext commit, use hardcode now

Comment (csoka) [reply src/plugins/intel_npu/src/backend/src/zero_backend.cpp:33]: Looks good, cheers

**Pull request merged**